plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -99,6 +100,16 @@ func RegisterFitPredicate(name string, predicate algorithm.FitPredicate) string
 	return RegisterFitPredicateFactory(name, func(PluginFactoryArgs) algorithm.FitPredicate { return predicate })
 }
 
+// RegisterRequiredFitPredicate registers a fit predicate with the algorithm registry,
+// it's always included in configuration. Returns the name with which the predicate was registered.
+func RegisterRequiredFitPredicate(name string, predicate algorithm.FitPredicate) string {
```

Comment:
I am not sure about this either. The way this method is written implies that this is a new first class way of adding "required predicates". I would remove this method and `requiredFitPredicateMap` and would add this predicate to the `predicates` map hard-codedly in `getFitPredicateFunctions` with a TODO that this hard-coded predicate is there for backward compatibility and should be removed in a couple of releases.